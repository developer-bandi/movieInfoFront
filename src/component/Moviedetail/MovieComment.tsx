/*eslint-disable*/

import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import styled from 'styled-components';
import { BsXLg } from 'react-icons/bs';
interface Props {
  uploadComment: (e: { type: string; key?: string }) => void;
  settingComment: (e: React.ChangeEvent<HTMLInputElement>) => void;
  deleteComment: (index: number) => void;
  commentList: { comment: string; date: string }[];
}
const MovieComment = ({
  uploadComment,
  settingComment,
  deleteComment,
  commentList,
}: Props) => {
  return (
    <MovieCommentBlock>
      <InputBox>
        <CommentInput onChange={settingComment} onKeyPress={uploadComment} />
        <UploadButton onClick={uploadComment}>등록</UploadButton>
      </InputBox>
      <CommentListBlock>
        {commentList.map((data, index) => {
          if (data.date === 'default') {
            return null;
          }
          return (
            <CommentBlock key={index}>
              <Comment>{data.comment}</Comment>
              <BottomBlock>
                <CommentDate>{data.date}</CommentDate>
                <DeleteButton
                  onClick={(e) => {
                    deleteComment(index);
                  }}
                />
              </BottomBlock>
            </CommentBlock>
          );
        })}
      </CommentListBlock>
    </MovieCommentBlock>
  );
};

const MovieCommentBlock = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  background: ${(props) => props.theme.background};
`;

const InputBox = styled.div`
  margin-top: 3rem;
  display: flex;
  border: 1px solid black;
`;

const CommentInput = styled.input`
  width: 60rem;
  height: 3rem;
  border: 0 solid black;
  font-size: 20px;
  &:focus {
    outline: none;
  }
  @media screen and (max-width: 500px) {
    width: 15rem;
  }
`;

const UploadButton = styled.div`
  width: 5rem;
  height: 3.2rem;
  background: #e6e6e6;

  &:hover {
    background: #c0c0c0;
  }
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 15px;
`;

const CommentListBlock = styled.div`
  margin-top: 2rem;
  margin-bottom: 2rem;
  width: 65rem;
  @media screen and (max-width: 500px) {
    width: 90%;

    padding: 5%;
  }
`;

const CommentBlock = styled.div`
  border-bottom: 0.5px solid black;
  margin-top: 0.5rem;
`;
const Comment = styled.div`
  font-size: 15px;
  margin-bottom: 0.5rem;
`;

const CommentDate = styled.div`
  font-size: 10px;
  margin-bottom: 0.5rem;
`;
const BottomBlock = styled.div`
  display: flex;
`;

const DeleteButton = styled(BsXLg)`
  width: 0.7rem;
  height: 0.7rem;
  margin-left: 0.5rem;
  &:hover {
    color: gray;
  }
`;

export default MovieComment;
