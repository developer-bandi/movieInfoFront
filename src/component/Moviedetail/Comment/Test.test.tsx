import { fireEvent, render, screen } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import useUploadComment from './Hook';
import MovieComment from './Presentational';
const MovieCommentApiData = {
  id: 1,
  content: 'test',
  createdAt: new Date(),
  User: {
    id: 1,
    nick: 'test',
  },
};
const MovieCommentState = {
  content: [MovieCommentApiData],
  loading: false,
  error: false,
};
describe('MovieComment Presentational', () => {
  it('전체 화면 검증', () => {
    const deleteCommentMock = jest.fn();
    const utils = render(
      <MovieComment
        uploadComment={jest.fn()}
        settingComment={jest.fn()}
        deleteComment={deleteCommentMock}
        commentList={MovieCommentState}
        loginedUserId={1}
      />
    );
    expect(utils.container).toMatchSnapshot();
    const deleteButton = screen.getByTestId('deleteButton');
    fireEvent.click(deleteButton);
    expect(deleteCommentMock).toBeCalledTimes(1);
  });
});

describe('MovieComment Hook', () => {
  const mockStore = configureMockStore();
  const store = mockStore({
    user: {
      content: { id: 1, userid: 'testId', nick: 'testnick' },
      loading: false,
      error: false,
    },
  });
  const wrapper = ({ children }: { children: React.FC }) => (
    <Provider store={store}>{children}</Provider>
  );

  const newmockStore = configureMockStore();
  const notLoginstore = newmockStore({
    user: {
      loading: false,
      error: false,
    },
  });
  const notLoginwrapper = ({ children }: { children: React.FC }) => (
    <Provider store={notLoginstore}>{children}</Provider>
  );

  it('로그인 클릭', async () => {
    const { result } = renderHook(() => useUploadComment('test'), {
      wrapper,
    });
    result.current({ type: 'click', key: undefined });
    expect(store.getActions()[0]).toStrictEqual({
      type: 'movieComment/addMovieComment',
      payload: { movieId: undefined, content: 'test' },
    });
  });
  it('로그인 엔터키', async () => {
    const { result } = renderHook(() => useUploadComment('test'), {
      wrapper,
    });
    result.current({ type: 'keypress', key: 'Enter' });
    expect(store.getActions()[1]).toStrictEqual({
      type: 'movieComment/addMovieComment',
      payload: { movieId: undefined, content: 'test' },
    });
  });
  it('비로그인 클릭', async () => {
    window.alert = jest.fn();
    const { result } = renderHook(() => useUploadComment('test'), {
      wrapper: notLoginwrapper,
    });
    result.current({ type: 'click', key: undefined });
    expect(window.alert).toBeCalledTimes(1);
  });
  it('비로그인 엔터', async () => {
    window.alert = jest.fn();
    const { result } = renderHook(() => useUploadComment('test'), {
      wrapper: notLoginwrapper,
    });
    result.current({ type: 'keypress', key: 'Enter' });
    expect(window.alert).toBeCalledTimes(1);
  });
});
