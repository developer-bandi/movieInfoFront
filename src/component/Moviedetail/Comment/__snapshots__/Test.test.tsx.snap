// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieComment/> Presentational 스냅샷 1`] = `
<div>
  <div
    class="sc-bdvvtL euTXYv"
  >
    <div
      class="sc-gsDKAQ idYvNZ"
    >
      <div
        class="sc-dkPtRN kcyxwW"
      >
        <div
          class="sc-hKwDye lhRTGn"
        >
          <div
            class="sc-eCImPb hZjPQD"
          >
            test
          </div>
          <div
            class="sc-jRQBWg ikjKwU"
          >
            방금전
          </div>
          <svg
            class="sc-gKclnd kEgghG"
            data-testid="deleteButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.293 1.293a1 1 0 0 1 1.414 0L8 6.586l5.293-5.293a1 1 0 1 1 1.414 1.414L9.414 8l5.293 5.293a1 1 0 0 1-1.414 1.414L8 9.414l-5.293 5.293a1 1 0 0 1-1.414-1.414L6.586 8 1.293 2.707a1 1 0 0 1 0-1.414z"
            />
          </svg>
        </div>
        <div
          class="sc-iCfMLu hduNzp"
        >
          test
        </div>
      </div>
    </div>
    <div
      class="sc-furwcr kCpfWk"
    >
      <textarea
        class="sc-pVTFL QgQDl"
      />
      <div
        class="sc-jrQzAO ejZmHN"
      >
        등록
      </div>
    </div>
  </div>
</div>
`;

exports[`MovieComment Presentational 전체 화면 검증 1`] = `
<div>
  <div
    class="sc-bdvvtL euTXYv"
  >
    <div
      class="sc-gsDKAQ idYvNZ"
    >
      <div
        class="sc-dkPtRN kcyxwW"
      >
        <div
          class="sc-hKwDye lhRTGn"
        >
          <div
            class="sc-eCImPb hZjPQD"
          >
            test
          </div>
          <div
            class="sc-jRQBWg ikjKwU"
          >
            방금전
          </div>
          <svg
            class="sc-gKclnd kEgghG"
            data-testid="deleteButton"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1.293 1.293a1 1 0 0 1 1.414 0L8 6.586l5.293-5.293a1 1 0 1 1 1.414 1.414L9.414 8l5.293 5.293a1 1 0 0 1-1.414 1.414L8 9.414l-5.293 5.293a1 1 0 0 1-1.414-1.414L6.586 8 1.293 2.707a1 1 0 0 1 0-1.414z"
            />
          </svg>
        </div>
        <div
          class="sc-iCfMLu hduNzp"
        >
          test
        </div>
      </div>
    </div>
    <div
      class="sc-furwcr kCpfWk"
    >
      <textarea
        class="sc-pVTFL QgQDl"
      />
      <div
        class="sc-jrQzAO ejZmHN"
      >
        등록
      </div>
    </div>
  </div>
</div>
`;
